monitor.py

```diff
@@ -1306,16 +1333,26 @@ def parse_entry(self, line: bytes) -> MutableMapping[str, Any]:  # noqa: C901, C
 
             elif event_type == 'navroute':
                 # Added in ED 3.7 - multi-hop route details in NavRoute.json
-                with open(join(self.currentdir, 'NavRoute.json'), 'rb') as rf:  # type: ignore
-                    try:
-                        entry = json.load(rf)
+                if (nv_json := self._parse_navroute_file()) is not None:
+                    entry_time = mktime(strptime(nv_json['timestamp'], '%Y-%m-%dT%H:%M:%SZ'))
```

Comment:
This also needs some paranoia.  Don't just assume `timestamp` is present in the file.  I don't trust Frontier *at all* at this point.  They might give us a file with literally just `{}` in it.